internal/Get-DbaADObject.ps1

```diff
@@ -24,108 +29,65 @@ This program is distributed in the hope that it will be useful, but WITHOUT ANY
-			}
-			return $dn
+		try {
+			Add-Type -AssemblyName System.DirectoryServices.AccountManagement
+		} catch {
+			write-warning 'Failed to load the required module'
 		}
-		try
-		{
-			$alldomains = @()
-			$currentforest = [System.DirectoryServices.ActiveDirectory.Forest]::GetCurrentForest()
-			$alldomains += $currentforest.Domains.name
-
-			$cd = $currentforest.Domains
-		}
-		catch
-		{
-			Write-warning "No Active Directory domains Found."
-			break
-		}
-		foreach ($domain in $cd)
-		{
-			try
-			{
-				$alldomains += ($Domain.GetAllTrustRelationships()).TargetName
+		switch ($Type) {
+			"User" {
+				$searchClass = [System.DirectoryServices.AccountManagement.UserPrincipal]
 			}
-			catch
-			{
-				$alldomains = $alldomains | Where-Object { $_ -ne $domain.name }
-				Write-Warning "Couldn't contact $domain"
+			"Group" {
+				$searchClass = [System.DirectoryServices.AccountManagement.GroupPrincipal]
 			}
-		}
-
-		$alldomains = $alldomains | Select-Object -Unique
-
-		# should leverage "process" cache for this, maybe @fred can help
-		$script:domains = @()
-
-		function Resolve-ExpensiveDomain([string]$NetBiosName) {
-			foreach ($domain in $alldomains) {
-				if ($script:domains.DNS -contains $domain) { continue }
-				try {
-					$dn = ConvertTo-Dn $domain
-					$translate = New-Object -comObject NameTranslate
-					$reflection = $translate.GetType()
-					$reflection.InvokeMember("Init", "InvokeMethod", $Null, $translate, (3, $Null)) | out-null
-					$reflection.InvokeMember("Set", "InvokeMethod", $Null, $translate, (1, $dn)) | out-null
-					$netbios = $reflection.InvokeMember("Get", "InvokeMethod", $Null, $translate, 3).Trim("\")
-					$script:domains += [pscustomobject]@{
-						DNS = $domain
-						DN = $dn
-						NetBios = $netbios
-						LDAP = "LDAP://" + $netbios + "/" + $DN
-					}
-					if ($NetBiosName -eq $netbios) {
-						break
-					}
-				} catch {
-					Write-Warning "Removing $domain from domain list"
-				}
+			"Computer" {
+				$searchClass = [System.DirectoryServices.AccountManagement.ComputerPrincipal]
+			}
+			default {
+				$searchClass = [System.DirectoryServices.AccountManagement.Principal]
 			}
 		}
-
 	}
 	PROCESS {
-		foreach($adobj in $ADObject) {
-			$domain, $filter = $adobj.Split("\")
-			if ($env:USERDOMAIN -eq $domain) {
-				$searcher = New-Object System.DirectoryServices.DirectorySearcher
-				$searcher.Filter = $filter
-			} else {
-				Resolve-ExpensiveDomain -NetBiosName $domain
-				$LDAP = ($script:domains | Where-Object NetBios -eq $domain).LDAP
-				$ad = New-Object System.DirectoryServices.DirectoryEntry $LDAP
-				$searcher = New-Object System.DirectoryServices.DirectorySearcher
-				$searcher.SearchRoot = $ad
-				$searcher.Filter = $filter
+		foreach($ADObj in $ADObject) {
+			$Splitted = $ADObj.Split("\")
+			if ($Splitted.Length -ne 2) {
+				Stop-Function -Message "You need to pass ADObject in DOMAIN\object format" -Silent $Silent
+				continue
```

Comment:
Instead of calling the continue statement in the next line, use the `continue` parameter. Otherwise this code will never run